### PR TITLE
Docs/rename and replace some sections

### PR DESCRIPTION
**Pull request by avsakharov**
# Description

In this PR:
1. Documentation is renamed to API reference
2. Replaced examples and tutorials
3. Tutorials description is written
4. Actor is deleted from Readme
5. Contribution rules has been moved to Development
6. Toy script has been added to the doc
7. Added description to community page
8. Added description to development page
9. Add description for each stage of ActorStage in types.py

# Checklist

- [x] I have covered the code with tests
- [x] I have added comments to my code to help others understand it
- [x] I have updated the documentation to reflect the changes
- [x] I have performed a self-review of the changes

### CODE REVIEW

**Comment by avsakharov** [comment]
Unfortunately, HAPPY_PATH and TOY_SCRIPT don't look pretty. And I don't know hot to fix it.
image

**Review by kudep — CHANGES_REQUESTED** [review]
conflicts

File: docs/source/tutorials.rst [hunk 1]
```diff
@@ -1,4 +1,25 @@
...
+The Pipeline section teaches the basics of the pipeline concept, how to use pre- and postprocessors,
+asynchronous groups and services, custom messenger interfaces, and extra handlers and extensions.
+The Script section covers the basics of the script concept, including conditions, responses, transitions,
+and serialization. It also includes tutorials on pre-response and pre-transitions processing.
+Finally, the Utils section covers the cache and LRU cache utilities in DFF.
+
+The main difference between Tutorials and Examples is that Examples typically show how to implement
+a specific feature or solve a particular problem, whereas Tutorials provide a more
+comprehensive overview of how to build a complete application.
+
```
Comment (kudep) [inline docs/source/tutorials.rst:20]: What about link to the `repository <https://github.com/deeppavlov/dialog_flow_demo>`_.

File: docs/source/tutorials/pipeline/1_basics.py [hunk 2]
```diff
@@ -0,0 +1 @@
+/home/avsakharov/deeppavlov/dialog_flow_framework/tutorials/pipeline/1_basics.py
```
Comment (kudep) [inline docs/source/tutorials/pipeline/1_basics.py:1]: what is it ? This one absolute path exists only for avsakharov

File: docs/source/tutorials/utils/1_cache.py [hunk 3]
```diff
@@ -0,0 +1 @@
+/home/avsakharov/deeppavlov/dialog_flow_framework/tutorials/utils/1_cache.py
```
Comment (kudep) [inline docs/source/tutorials/utils/1_cache.py:1]: Absolute path

File: docs/source/tutorials/context_storages/dbs/file.json [hunk 4]
```diff
@@ -0,0 +1 @@
..., "node1"]}, "requests": {"0": {"text": "Hi", "commands": null, "attachments": null, "annotations": null, "misc": null}, "1": {"text": "i'm fine, how are you?", "commands": null, "attachments": null, "annotations": null, "misc": null}, "2": {"text": "Let's talk about music.", "commands": null, "attachments": null, "annotations": null, "misc": null}, "3": {"text": "Ok, goodbye.", "commands": null, "attachments": null, "annotations": null, "misc": null}, "4": {"text": "Hi", "commands": null, "attachments": null, "annotations": null, "misc": null}}, "responses": {"0": {"text": "Hi, how are you?", "commands": null, "attachments": null, "annotations": null, "misc": null}, "1": {"text": "Good. What do you want to talk about?", "commands": null, "attachments": null, "annotations": null, "misc": null}, "2": {"text": "Sorry, I can not talk about music now.", "commands": null, "attachments": null, "annotations": null, "misc": null}, "3": {"text": "bye", "commands": null, "attachments": null, "annotations": null, "misc": null}, "4": {"text": "Hi, how are you?", "commands": null, "attachments": null, "annotations": null, "misc": null}}, "misc": {}, "validation": false, "framework_states": {}}}
```
Comment (kudep) [inline docs/source/tutorials/context_storages/dbs/file.json:1]: What is this file needed for?

**Review by kudep — CHANGES_REQUESTED** [review]

File: docs/source/tutorials/context_storages/dbs/file.json (hunk 4 above)
Comment (avsakharov) [reply docs/source/tutorials/context_storages/dbs/file.json]: Deleted

File: docs/source/tutorials.rst (hunk 1 above)
Comment (avsakharov) [reply docs/source/tutorials.rst:20]: The link has been added to the Examples section. Would it be better to move it to Tutorials?

File: docs/source/tutorials/pipeline/1_basics.py (hunk 2 above)
Comment (avsakharov) [reply docs/source/tutorials/pipeline/1_basics.py]: Deleted

File: docs/source/tutorials/utils/1_cache.py (hunk 3 above)
Comment (avsakharov) [reply docs/source/tutorials/utils/1_cache.py]: Deleted

**Comment by avsakharov** [comment]
> I found [these](https://stackoverflow.com/a/48578508) [two](https://stackoverflow.com/q/27875455) solutions on the internet. The second one contains even more than one answer actually. Maybe any of them can help?

Solution from first link gives the same result. The second solution, unfortunately, I was not able to reproduce. Can you, please, try to implement it?

**Comment by avsakharov** [comment]
By the way, this warning: "dialog_flow_framework/docs/source/release_notes.rst: WARNING: document isn't included in any toctree" is corrected in PR 92.

File: dff/script/core/types.py [hunk 5]
```diff
@@ -40,39 +40,56 @@
...
     """
 
-    #: This stage is used for the context initializing.
+    #: This stage is used for the context initialization.
+    #: It involves setting up the conversation context.
     CONTEXT_INIT = auto()
 
-    #: This stage is used to get the previous node.
+    #: This stage is used to retrieve the previous node.
+    #: It involves identifying the previous node in the conversation history.
```
Comment (RLKRo) [inline dff/script/core/types.py:56]: Isn't it just repetition?

File: dff/script/core/types.py [hunk 6]
```diff
@@ -40,39 +40,56 @@
...
     GET_PREVIOUS_NODE = auto()
 
-    #: This stage is used for rewriting the previous node.
+    #: This stage is used to rewrite the previous node.
+    #: It involves updating the previous node in the conversation history
+    #: to reflect any changes made during the current conversation turn.
     REWRITE_PREVIOUS_NODE = auto()
 
-    #: This stage is used for running pre-transitions processing.
+    #: This stage is used for running pre-response processing.
```
Comment (RLKRo) [inline dff/script/core/types.py:64]: Why this change? It is used for pre-transition processing.

File: dff/script/core/types.py [hunk 7]
```diff
@@ -40,39 +40,56 @@
...
 
-    #: This stage is used to get true labels.
+    #: This stage is used to retrieve the true labels.
+    #: It involves determining the correct label to take based
+    #: on the current conversation context.
     GET_TRUE_LABELS = auto()
 
-    #: This stage is used to get next node.
+    #: This stage is used to retrieve the next node.
+    #: It involves identifying the next node in the conversation flow.
```
Comment (RLKRo) [inline dff/script/core/types.py:74]: Again, this is repetitive. Maybe just `This stage is used to retrieve the next node in the conversation flow.`?

File: dff/utils/testing/cleanup_db.py [hunk 8]
```diff
@@ -20,37 +26,67 @@
 
 
 async def delete_json(storage: JSONContextStorage):
+    """
+    Deletes all data from a JSON context storage.
```
Comment (RLKRo) [inline dff/utils/testing/cleanup_db.py:30]: Use `Delete` instead of `Deletes` here and in other functions.
[PEP-257](https://peps.python.org/pep-0257/):
> The docstring is a phrase ending in a period. It prescribes the function or method’s effect as a command (“Do this”, “Return that”), not as a description; e.g. don’t write “Returns the pathname …”.

File: dff/utils/testing/common.py [hunk 9]
```diff
@@ -1,3 +1,8 @@
+"""
+Common
+------
+This module contains several functions which is used to run demonstrations in tutorials.
```
Comment (RLKRo) [inline dff/utils/testing/common.py:4]:
```suggestion
This module contains several functions which are used to run demonstrations in tutorials.
```

File: dff/utils/testing/toy_script.py [hunk 10]
```diff
@@ -1,6 +1,13 @@
+"""
+Toy script
+----------
+This module contains a simple :py:class:`~.Script` and dialog which are used
```
Comment (RLKRo) [inline dff/utils/testing/toy_script.py:4]:
```suggestion
This module contains a simple :py:class:`~.Script` and a dialog which are used
```

File: dff/utils/testing/toy_script.py [hunk 11]
```diff
@@ -1,6 +1,13 @@
...
+in tutorials.
+"""
 from dff.script.conditions import exact_match
 from dff.script import TRANSITIONS, RESPONSE, Message
 
+#:
```
Comment (RLKRo) [inline dff/utils/testing/toy_script.py:10]: Why add an empty docstring?

File: dff/script/core/types.py [hunk 12]
```diff
@@ -40,39 +40,56 @@
...
     The class which holds keys for the handlers. These keys are used
-    for the actions of :py:class:`~dff.script.Actor`.
+    for the actions of :py:class:`.Actor`. Each stage represents
+    a specific step in the conversation flow. Here is a brief description
+    of each stage.
     """
 
-    #: This stage is used for the context initializing.
+    #: This stage is used for the context initialization.
+    #: It involves setting up the conversation context.
```
Comment (RLKRo) [inline dff/script/core/types.py:52]: Why is a different docstring style used here?
The rest of the library uses `"""""`.

File: docs/source/examples.rst [hunk 13]
```diff
@@ -1,8 +1,4 @@
 Examples
 --------
 
-.. toctree::
-   :name: examples
-   :glob:
-
-   examples/*/index
+Work in progress... Examples will be available in this `repository <https://github.com/deeppavlov/dialog_flow_demo>`_.
```
Comment (RLKRo) [inline docs/source/examples.rst:4]: They are already available though.
```suggestion
Work in progress... Examples are available in this `repository <https://github.com/deeppavlov/dialog_flow_demo>`_.
```

File: docs/source/tutorials.rst [hunk 14]
```diff
@@ -1,4 +1,25 @@
 Tutorials
 ---------
+Tutorials page is a collection of instructional materials designed to help developers learn
+how to use DFF to build conversational agents. The tutorials cover a range of topics,
+from getting started with DFF to more advanced topics such as integrating external APIs.
+Each tutorial includes detailed explanations and code examples. Tutorials cover different aspects
+of the framework and organized into sections.
```
Comment (RLKRo) [inline docs/source/tutorials.rst:7]:
```suggestion
of the framework and are organized into sections.
```

File: docs/source/tutorials.rst [hunk 15]
```diff
@@ -1,4 +1,25 @@
...
 
-Work in progress... Examples of tutorials will be available in this `repository <https://github.com/deeppavlov/dialog_flow_demo>`_.
\ No newline at end of file
+The Context Storages section describes how to use context stores in DFF.
```
Comment (RLKRo) [inline docs/source/tutorials.rst:9]:
```suggestion
The Context Storages section describes how to use context storages in DFF.
```

File: .gitignore [hunk 16]
```diff
@@ -4,7 +4,8 @@ dist/
 venv/
 build/
 docs/source/apiref
-docs/source/examples
+docs/source/release_notes.rst
+docs/source/tutorials
```
Comment (RLKRo) [inline .gitignore:8]: Why do we ignore these two? Are they auto generated?

**Review by RLKRo — CHANGES_REQUESTED** [review]

File: .gitignore (hunk 16 above)
Comment (kudep) [reply .gitignore:8]: yes

Comment (kudep) [reply .gitignore:8]: I think we can move auto-generated content to a special directory @pseusys

File: dff/context_storages/protocol.py [hunk 17]
```diff
@@ -22,6 +22,12 @@
 
 
 def get_protocol_install_suggestion(protocol_name: str) -> str:
+    """
+    This function is used to provide suggestions for installing
+    the necessary dependencies for a specific protocol.
```
Comment (RLKRo) [inline dff/context_storages/protocol.py:27]:
```suggestion
    Provide suggestions for installing the necessary dependencies for a specific protocol.
```

File: dff/pipeline/service/extra.py [hunk 18]
```diff
@@ -168,6 +168,18 @@ def info_dict(self) -> dict:
...
+    """
+    A handler for extra functions that are executed before the component's main function.
+
+    :param functions: A callable or a list of callables that will be executed
+        before the component's main function.
+    :type functions: ExtraHandlerBuilder
+    :param timeout: Optional timeout for the execution of the extra functions, in
+        seconds.
+    :param asynchronous: Optional flag that indicates whether the extra functions
+        should be executed asynchronously.
```
Comment (RLKRo) [inline dff/pipeline/service/extra.py:180]: Mention that the default value of the flag is True if all the functions in this handler are asynchronous.

File: dff/utils/testing/toy_script.py [hunk 19]
```diff
@@ -1,3 +1,9 @@
+"""
+Toy script
+----------
+This module contains a simple :py:class:`~.Script` and a dialog which are used
```
Comment (RLKRo) [inline dff/utils/testing/toy_script.py:4]: What does ```
:py:class:`~.Script`
``` reference here? There is no class `Script`.

**Review by RLKRo — CHANGES_REQUESTED** [review]
